src/webui.c

```diff
@@ -77,6 +78,52 @@ fetch_wa_cookie(struct evhttp_request *req)
     return wa;
 }
 
+static void
+send_json_worker(struct evhttp_request *req, void *arg)
+{
+    struct evbuffer *buf = evhttp_request_get_output_buffer(req);
+    struct evkeyvalq *hdrs_out = NULL;
+    bool found = false;
+    double mh[6] = {0};
+    char rig_id[MAX_RIG_ID] = {0};
+    char wnbuf[128] = {0};
+    char *wn = NULL;
+    const char *url = evhttp_request_get_uri(req);
+    url += 8; 
+    strncpy(wnbuf, url, sizeof(wnbuf)-1);
+    wnbuf[strcspn(wnbuf, "/")] = 0;
+    wn = evhttp_uridecode(wnbuf, 1, NULL);
+    const char *wa = fetch_wa_cookie(req);
+    hdrs_out = evhttp_request_get_output_headers(req);
+    evhttp_add_header(hdrs_out, "Content-Type", "application/json");
+    if (!wn || !wa)
```

Comment:
There's a leak here. If `wn` is not NULL but `wa` is NULL, `wn` wont be freed. I suggest moving the call to `fetch_wa_cookie` to _before_ `evhttp_uridecode` and exiting there if no wallet address was supplied.

Reply:
moved to fetch wa higher and returning if wa is null